lib/explorer/query.ex

```diff
@@ -599,5 +599,30 @@ defmodule Explorer.Query do
     |> Enum.map(&%{Explorer.Shared.apply_impl(df, :pull, [&1]) | name: &1})
   end
 
+  defmacro select(do: clauses) do
+    conditions =
+      clauses
+      |> Enum.map(fn {:->, _, [[condition], on_true]} -> [condition, on_true] end)
+
+    quote do
+      import Explorer.Query
+
+      unquote(conditions)
+      |> Enum.reverse()
+      |> Enum.reduce(nil, fn [condition, truthy], acc ->
```

Comment:
Something you could optimize is to check if `condition == true`. If that's the case, you know it will always be true and you can skip the whole select. Something like this:

```
  {true, truthy}, _acc -> Explorer.Shared.lazy_series!(truthy)
  {condition, truthy}, acc -> ...
```